drivers/input/input_esp32_touch_sensor.c

```diff
@@ -0,0 +1,349 @@
+/*
+ * Copyright (c) 2023 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT espressif_touch_sensor
+
+#include <zephyr/device.h>
+#include <zephyr/input/input.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#include <esp_err.h>
+#include <soc/soc_pins.h>
+#include <soc/periph_defs.h>
+#include <hal/touch_sensor_types.h>
+#include <hal/touch_sensor_hal.h>
+#include <driver/rtc_io.h>
+#include <esp_intr_alloc.h>
+
+LOG_MODULE_REGISTER(espressif_touch_sensor, CONFIG_INPUT_LOG_LEVEL);
+
+#if defined(CONFIG_COUNTER_RTC_ESP32)
+#     error "Conflict detected: COUNTER_RTC_ESP32 enabled"
```

Comment:
nit: can you keep the macro untabbed? `#error` etc... I imagine that's the style you use in IDF or something but it's very uncommon in zephyr and it looks a bit weird :-)